toolchain/check/facet_type.cpp

```diff
@@ -165,18 +165,16 @@ auto InitialFacetTypeImplWitness(
     }
 
     if (table_entry != SemIR::ImplWitnessTablePlaceholder::SingletonInstId) {
-      if (context.constant_values().Get(table_entry) != rewrite_value) {
-        // TODO: Figure out how to print the two different values
-        // `const_id` & `rewrite_value` in the diagnostic
-        // message.
-        CARBON_DIAGNOSTIC(AssociatedConstantWithDifferentValues, Error,
-                          "associated constant {0} given two different values",
-                          SemIR::NameId);
+      if (table_entry != rewrite_value) {
+        CARBON_DIAGNOSTIC(
+            AssociatedConstantWithDifferentValues, Error,
+            "associated constant {0} given two different values {1} and {2}",
+            SemIR::NameId, InstIdAsType, InstIdAsType);
```

Comment:
These `InstId`s need not be types. What happens if you have a test like
```
interface I { let N:! i32; }
impl () as I where .N = 1 and .N = 2 {}
```

Reply:
We should really rename `InstIdAsType` / `stringify_type` -- they can actually stringify any constant. (This is necessary because they can stringify *arguments* to generic types.)

Reply:
Moved the diagnostic changes out to #5285.